service.languagepreferencemanager/resources/lib/prefutils.py

```diff
@@ -0,0 +1,369 @@
+        test = subName.lower()
+        matches = ['forced', 'forcés']
+        found = any(x in test for x in matches)
+        return ((forced == 'false') and not found) or ((forced == 'true') and found)
+
+    def isExternalSub(self, subName):
+        test = subName.lower()
+        matches = ['ext']
+        return any(x in test for x in matches)
+    
+    def getDetails(self):
+        activePlayers ='{"jsonrpc": "2.0", "method": "Player.GetActivePlayers", "id": 1}'
+        json_query = xbmc.executeJSONRPC(activePlayers)
+        #json_query = unicode(json_query, 'utf-8', errors='ignore')
+        json_response = simplejson.loads(json_query)
+        activePlayerID = json_response['result'][0]['playerid']
+        details_query_dict = {  "jsonrpc": "2.0",
+                                "method": "Player.GetProperties",
+                                "params": { "properties": 
+                                            ["currentaudiostream", "audiostreams", "subtitleenabled",
+                                             "currentsubtitle", "subtitles" ],
+                                            "playerid": activePlayerID },
+                                "id": 1}
+        details_query_string = simplejson.dumps(details_query_dict)
+        json_query = xbmc.executeJSONRPC(details_query_string)
+        #json_query = unicode(json_query, 'utf-8', errors='ignore')
+        json_response = simplejson.loads(json_query)
+        
+        if 'result' in json_response and json_response['result'] != None:
+            self.selected_audio_stream = json_response['result']['currentaudiostream']
+            self.selected_sub = json_response['result']['currentsubtitle']
+            self.selected_sub_enabled = json_response['result']['subtitleenabled']
+            self.audiostreams = json_response['result']['audiostreams']
+            self.subtitles = json_response['result']['subtitles']
+        log(LOG_DEBUG, json_response )
+        
+        if (not settings.custom_condsub_prefs_on and not settings.custom_audio_prefs_on and not settings.custom_sub_prefs_on):
+            log(LOG_DEBUG, 'No custom prefs used at all, skipping extra Video tags/genres JSON query.')
+            self.genres_and_tags = set()
+            return        
+        
+        genre_tags_query_dict = {"jsonrpc": "2.0",
+                                 "method": "Player.GetItem",
+                                 "params": { "properties":
+                                            ["genre", "tag"],
+                                            "playerid": activePlayerID },
+                                 "id": 1}
+        genre_tags_query_string = simplejson.dumps(genre_tags_query_dict)
+        json_query = xbmc.executeJSONRPC(genre_tags_query_string)
+        #json_query = unicode(json_query, 'utf-8', errors='ignore')
+        json_response = simplejson.loads(json_query)
+        if 'result' in json_response and json_response['result'] != None:
```

Comment:
`is not None`

Reply:
Not mandatory hence will be fixed in next update.